odbc/src/api/cancel_tests.rs

```diff
@@ -0,0 +1,140 @@
+use crate::{handles::definitions::*, SQLCancel};
+use definitions::SqlReturn;
+use mongo_odbc_core::MongoConnection;
+use mongodb::sync::Client;
+use std::env;
+
+mod integration {
```

Comment:
Yes, I agree a full integration test would be useful, though I'm a bit unsure of if/how it would actually be possible if we can't use the same handle amongst multiple threads. I had tried a couple ways but couldn't quite get it to work. Do you have a certain test setup in mind?

Reply:
uhm, no. I quickly looked into it and it looks like, we would need some kind of thread-safe wrapper around the handle. I am not a Rust multi-thread expert at all unfortunately.

Reply:
Likewise, though I'm not sure it will be trivial. Maybe we can file a ticket and it could be a good skunkworks project? I do think its nice to have and worth implementing, but none of the tools we're currently focussed on rely on this functionality

Reply:
Yes, this can be a follow up item.